samples/benchmarks/coremark/src/main.c

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/logging/log_ctrl.h>
+#include <zephyr/drivers/gpio.h>
+
+#include "coremark_zephyr.h"
+
+LOG_MODULE_REGISTER(app, LOG_LEVEL_INF);
+
+/*
+ * Get button and led configuration from the devicetree.
+ * Application expects to have button and led aliases present in devicetree.
+ */
+#define BUTTON_NODE             DT_ALIAS(button)
+#define LED_NODE                DT_ALIAS(led)
+
+#if !DT_NODE_HAS_STATUS(BUTTON_NODE, okay)
+	#error "Unsupported board: /"button/" alias is not defined."
+#endif
+
+#if !DT_NODE_HAS_STATUS(LED_NODE, okay)
+	#error "Unsupported board: /"led/" alias is not defined."
+#endif
+
+#define BUTTON_LABEL            DT_PROP(DT_ALIAS(button), label)
+
+static K_SEM_DEFINE(start_coremark, 0, 1);
+static bool coremark_in_progress;
+
+static const struct gpio_dt_spec start_button = GPIO_DT_SPEC_GET(BUTTON_NODE, gpios);
+static const struct gpio_dt_spec status_led = GPIO_DT_SPEC_GET(LED_NODE, gpios);
+
+static struct gpio_callback button_cb_data;
+
```

Comment:
Will do. Thanks for advice!

Reply:
Frankly, I'd suggest to remove that code at all for the sake of simplicity. The point is CoreMark is a benchmark for a CPU and not really dependent on any peripherals. That's why it's so popular - it's super portable. But adding all these bells and whistles we complicate code and limit its usability.
Moreover, why someone really needs LED indication of the test execution? And if there's really some use of that, I'd suggest to move that code to the wrapper code so that all tests could use that functionality.

Reply:
Buttons and LED is a simple way to control test externally.
There is CM (iterations/sec) and CM/MHz values, and no buttons/leds etc required to evaluate it.
But there is another one value - CM/mA. And to provide this test you must setup some external equipment (amperemeter etc.) and measure current consumption only for test execution time.
In this case you can trigger test start with button and do measurements only while led indicates ```test in progress```.

From human point of view - button and led. From automation point of view - that is simple gpio signals for triggering the test and tracking ```test in progress``` state.

So, I would prefer to leave it and provide this external control. We can use flash & run as a default configuration and do not use any leds and gpios - so it would be possible to execute it on any board.